diff --git a/drivers/amvdec_ports/aml_vcodec_vfm.c b/drivers/amvdec_ports/aml_vcodec_vfm.c
index 36c1820..8ce40a6 100644
--- a/drivers/amvdec_ports/aml_vcodec_vfm.c
+++ b/drivers/amvdec_ports/aml_vcodec_vfm.c
@@ -57,8 +57,6 @@ static void vdec_vf_put(struct vframe_s *vf, void *op_arg)
 		return;
 	}
 
-	atomic_set(&vf->use_cnt, 1);
-
 	vfq_push(&vfm->vf_que_recycle, vf);
 
 	/* schedule capture work. */
@@ -95,7 +93,7 @@ void video_vf_put(char *receiver, struct vdec_fb *fb, int id)
 	aml_v4l2_debug(2, "[%d] TO   (%s) vf: %p, idx: %d",
 		id, vfp->name, vf, vf->index);
 
-	if (vfp && vf && atomic_dec_and_test(&vf->use_cnt))
+	if (vfp && vf)
 		vf_put(vf, receiver);
 }
 
diff --git a/drivers/amvdec_ports/decoder/vdec_h264_if.c b/drivers/amvdec_ports/decoder/vdec_h264_if.c
index efa9fc7..79edcfa 100644
--- a/drivers/amvdec_ports/decoder/vdec_h264_if.c
+++ b/drivers/amvdec_ports/decoder/vdec_h264_if.c
@@ -614,8 +614,6 @@ static void vdec_h264_get_vf(struct vdec_h264_inst *inst, struct vdec_fb **out)
 		return;
 	}
 
-	atomic_set(&vf->use_cnt, 1);
-
 	aml_vcodec_debug(inst, "%s() [%d], vf: %p, v4l_mem_handle: %lx, idx: %d\n",
 		__func__, __LINE__, vf, vf->v4l_mem_handle, vf->index);
 
diff --git a/drivers/amvdec_ports/decoder/vdec_hevc_if.c b/drivers/amvdec_ports/decoder/vdec_hevc_if.c
index 0870da7..5dc404e 100644
--- a/drivers/amvdec_ports/decoder/vdec_hevc_if.c
+++ b/drivers/amvdec_ports/decoder/vdec_hevc_if.c
@@ -505,8 +505,6 @@ static void vdec_hevc_get_vf(struct vdec_hevc_inst *inst, struct vdec_fb **out)
 		return;
 	}
 
-	atomic_set(&vf->use_cnt, 1);
-
 	aml_vcodec_debug(inst, "%s() [%d], vf: %p, v4l_mem_handle: %lx, idx: %d\n",
 		__func__, __LINE__, vf, vf->v4l_mem_handle, vf->index);
 
diff --git a/drivers/amvdec_ports/decoder/vdec_vp9_if.c b/drivers/amvdec_ports/decoder/vdec_vp9_if.c
index 757be84..24c367f 100644
--- a/drivers/amvdec_ports/decoder/vdec_vp9_if.c
+++ b/drivers/amvdec_ports/decoder/vdec_vp9_if.c
@@ -473,8 +473,6 @@ static void vdec_vp9_get_vf(struct vdec_vp9_inst *inst, struct vdec_fb **out)
 		return;
 	}
 
-	atomic_set(&vf->use_cnt, 1);
-
 	aml_vcodec_debug(inst, "%s() [%d], vf: %p, v4l_mem_handle: %lx, idx: %d\n",
 		__func__, __LINE__, vf, vf->v4l_mem_handle, vf->index);
 
